### PR TITLE
Add checks of public key for ECC and curve25519

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -3761,6 +3761,12 @@ int EccSharedSecret(WOLFSSL* ssl, ecc_key* priv_key, ecc_key* pub_key,
     else
 #endif
     {
+#ifdef OPENSSL_EXTRA
+        ret = wc_ecc_check_key(pub_key);
```
Comment (toddouska) [inline src/internal.c]: I'm having trouble following the logic of only checking an eco public key in OpenSSL compatibility mode.  Please explain.

File: src/internal.c [hunk 2]
```diff
@@ -25561,6 +25567,22 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
                                 }
                             }
 
+                            if ((ret = wc_curve25519_check_public(
+                                    input + args->idx, args->length,
+                                    EC25519_LITTLE_ENDIAN)) != 0) {
+                        #ifdef OPENSSL_EXTRA
```
Comment (toddouska) [inline src/internal.c]: Let's put verbose alerts in a separate define that OPENSSL_EXTRA may turn on, if that seems appropriate.

File: src/internal.c (hunk 1 above)
Comment (SparkiDev) [reply src/internal.c]: OK. DH has a WOLFSSL_VALIDATE_FFC_IMPORT define internally to check the DH public key.
Maybe a WOLFSSL_VALIDATE_ECC_IMPORT could be used?

File: src/internal.c (hunk 2 above)
Comment (SparkiDev) [reply src/internal.c]: I will make the change: WOLFSSL_EXTRA_ALRETS

File: src/internal.c (hunk 1 above)
Comment (SparkiDev) [reply src/internal.c]: The check is expensive.
I'll use WOLFSSL_VALIDATE_ECC_IMPORT and only turn it on for OPENSSL_ALL.

Comment (SparkiDev) [reply src/internal.c]: OK. No check OPENSSL_EXTRA || WOLFSSL_VALIDATE_ECC_IMPORT for both calls.

**Comment by SparkiDev** [comment]
retest this please.

**Comment by SparkiDev** [comment]
retest this please.

**Comment by SparkiDev** [comment]
Removed calls to wc_ecc_check_key as it is called in wc_ecc_import_x963_ex.

File: wolfcrypt/src/curve25519.c [hunk 3]
```diff
@@ -257,6 +257,64 @@ int wc_curve25519_import_public_ex(const byte* in, word32 inLen,
...
+ * pubSz   Size of public key in bytes.
+ * endian  Big-endian or little-endian.
+ * returns BAD_FUNC_ARGS when pub is NULL,
+ *         BUFFER_E when size of public key is zero;
+ *         ECC_OUT_OF_RANGE_E if the high bit is set;
+ *         ECC_BAD_ARG_E if key length is not 32 bytes, public key value is
+ *         zero or one; and
+ *         0 otherwise.
+ */
+int wc_curve25519_check_public(const byte* pub, word32 pubSz, int endian)
```
Comment (toddouska) [inline wolfcrypt/src/curve25519.c]: endian needs to be explained better.  If the underlying architecture is the meaning, have the user provide it is not acceptable.

Comment (SparkiDev) [reply wolfcrypt/src/curve25519.c]: The endian-ness is describing the data.
All import and export functions have a version that takes the endian of the data as a parameter.
Documentation updated.

Comment (toddouska) [reply wolfcrypt/src/curve25519.c]: Thanks.

**Pull request merged**